contracts/EventBasedPredictionMarket.sol

```diff
@@ -0,0 +1,135 @@
+    // to collateralPerPair and each long to 0. 1e18 makes each long worth collateralPerPair and short 0.
+    uint256 public expiryPercentLong;
+    bytes32 public priceIdentifier;
+
+    // Price returned from the Optimistic oracle at settlement time.
+    int256 public expiryPrice;
+
+    // External contract interfaces.
+    IERC20 public collateralToken;
+    ExpandedIERC20 public longToken;
+    ExpandedIERC20 public shortToken;
+    FinderInterface public finder;
+    LongShortPairFinancialProductLibrary public financialProductLibrary;
+
+    // Optimistic oracle customization parameters.
+    bytes public customAncillaryData;
+    uint256 public proposerReward;
+    uint256 public optimisticOracleLivenessTime;
+    uint256 public optimisticOracleProposerBond;
+
+    // Define the contract's constructor parameters as a struct to enable more variables to be specified.
+    struct ConstructorParams {
+        string pairName; // Name of the long short pair contract.
+        uint256 expirationTimestamp; // Unix timestamp of when the contract will expire.
+        uint256 collateralPerPair; // How many units of collateral are required to mint one pair of synthetic tokens.
+        bytes32 priceIdentifier; // Price identifier, registered in the DVM for the long short pair.
+        // bool enableEarlyExpiration; // Enables the LSP contract to be settled early.
+        ExpandedIERC20 longToken; // Token used as long in the LSP. Mint and burn rights needed by this contract.
+        ExpandedIERC20 shortToken; // Token used as short in the LSP. Mint and burn rights needed by this contract.
+        IERC20 collateralToken; // Collateral token used to back LSP synthetics.
+        LongShortPairFinancialProductLibrary financialProductLibrary; // Contract providing settlement payout logic.
+        bytes customAncillaryData; // Custom ancillary data to be passed along with the price request to the OO.
+        uint256 proposerReward; // Optimistic oracle reward amount, pulled from the caller of the expire function.
+        uint256 optimisticOracleLivenessTime; // OO liveness time for price requests.
+        uint256 optimisticOracleProposerBond; // OO proposer bond for price requests.
+        FinderInterface finder; // DVM finder to find other UMA ecosystem contracts.
+        address timerAddress; // Timer used to synchronize contract time in testing. Set to 0x000... in production.
+    }
+
+    constructor(ConstructorParams memory params) Testable(params.timerAddress) {
+        // During contract construction, set the contract's parameters.
+        // If the proposer reward was set then pull it from the caller of the function.
+        // requestPrice to OO
+        // setEventMarket
+        // Set the Optimistic oracle liveness for the price request.
+        // Set the Optimistic oracle proposer bond for the price request.
+
+        expirationTimestamp = getCurrentTime(); // Set the request timestamp to the current block timestamp.
```

Comment:
I'm using `getCurrentTime()` assuming this value is irrelevant when setting the request as an event based one.